src/compiler/factory.ts

```diff
@@ -699,6 +723,35 @@ namespace ts {
         return node;
     }
 
+    export function createWith(expression: Expression, statement: Statement, location?: TextRange): ReturnStatement {
+        const node = <WithStatement>createNode(SyntaxKind.WithStatement, location);
+        node.expression = expression;
+        node.statement = statement;
+        return node;
+    }
+
+    export function createThrow(expression: Expression, location?: TextRange): ReturnStatement {
+        const node = <ThrowStatement>createNode(SyntaxKind.ThrowStatement, location);
+        node.expression = expression;
+        return node;
+    }
+
+    export function createTryCatchFinally(tryBlock: Block, catchClause: CatchClause, finallyBlock: Block, location?: TextRange) {

```

Comment:
Since `catchClause` and `finallyBlock` are both optional/undefinable (as done below) - why not annotate them both with a `?`?